emperor/support_files/js/animations-controller.js

```diff
@@ -440,13 +443,16 @@ define([
     this.playing = false;
     this.director = null;
 
-    view.tubes.forEach(function(tube) {
-      if (tube.parent !== null) {
+    var allTubes = view.staticTubes.concat(view.dynamicTubes);
```

Comment:
This type of operation or pattern on both types of tubes happens a couple of times. Do you think we should have a `DecompositionView.getTubes` method that returns both types of tubes?

Reply:
Done